### ISSUE TITLE
Discrepancy between fmt and run commands in projects with multiple modules

### ISSUE THREAD

**Issue by nieomylnieja** [golangci/golangci-lint#6170]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Nix

### Description of the problem

When running `fmt` and `run` commands inside a project with multiple modules, the formmater formats paths in nested module, but the linting phase of `run` does not report formatting errors.
I've been only able to reproduce this with `gofumpt`.
I've created a simple project which reproduces this behavior, see below.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 2.5.0 built with go1.25.1 from v2.5.0 on 1970-01-01T00:00:00Z
```

</details>

### Configuration

<details>

```console
# yaml-language-server: $schema=https://golangci-lint.run/jsonschema/golangci.jsonschema.json
version: "2"
linters:
  default: none
formatters:
  enable:
    - gofumpt
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.25.2 linux/amd64
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO golangci-lint has version 2.5.0 built with go1.25.1 from v2.5.0 on 1970-01-01T00:00:00Z
INFO [config_reader] Config search paths: [./ /home/mh/projects/golangci-lint-reproduce /home/mh/projects /home/mh /home /]
INFO [config_reader] Used config file .golangci.yml
INFO [config_reader] Module name "github.com/nieomylnieja/bug"
INFO maxprocs: Leaving GOMAXPROCS=20: CPU quota undefined
INFO [goenv] Read go env for 5.037183ms: map[string]string{"GOCACHE":"/home/mh/.cache/go-build", "GOROOT":"/nix/store/k1kn1c59ss7nq6agdppzq3krwmi3xqy4-go-1.25.2/share/go"}
INFO [lintersdb] Active 1 linters: [gofumpt]
INFO [loader] Go packages loading at mode 8199 (files|name|compiled_files) took 50.299005ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 166.537µs
INFO [linters_context/goanalysis] analyzers took 300.294µs with top 10 stages: gofumpt: 275.069µs, typecheck: 25.225µs
INFO [runner] processing took 8.784µs with stages: max_same_issues: 2.368µs, exclusion_paths: 946ns, path_absoluter: 878ns, exclusion_rules: 877ns, diff: 370ns, path_prettifier: 360ns, source_code: 345ns, path_relativity: 344ns, max_from_linter: 341ns, nolint_filter: 325ns, filename_unadjuster: 288ns, max_per_file_from_linter: 211ns, path_shortener: 172ns, cgo: 159ns, invalid_issue: 153ns, sort_results: 152ns, fixer: 143ns, uniq_by_line: 124ns, generated_file_filter: 116ns, severity-rules: 112ns
INFO [runner] linters took 6.252045ms with stages: goanalysis_metalinter: 6.113308ms
0 issues.
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 2 samples, avg is 38.3MB, max is 40.3MB
INFO Execution took 62.332748ms

```

</details>

### A minimal reproducible example or link to a public repository

<details>

The reproduction minimal project is here: https://github.com/nieomylnieja/golangci-lint-formatting-issue-reproduction

The project has two modules, both are listed in go.work.
When running the linters with `run` command from the root of the project there are no issues reported, while running `fmt -d` command yields a diff with changed imports.
On the other hand, when we enter the nested module and execute `run`, the linter does detect a formatting issue.
This discrepancy is causing issues as I cannot rely on `run` command to verify the whole project for me.
In addition I tried reproducing this with `gofumpt` itself in version 0.9.1 (which I believe is the one used by 2.5.0) and I wasn't able to get anything at all, it didn't format the file even If I pointed it at it 🤷 

Run:

```shell
$ golangci-lint run
0 issues.
$ golangci-lint fmt -d
diff /home/mh/projects/golangci-lint-reproduce/examplemodule/main.go.orig /home/mh/projects/golangci-lint-reproduce/examplemodule/main.go
--- /home/mh/projects/golangci-lint-reproduce/examplemodule/main.go.orig
+++ /home/mh/projects/golangci-lint-reproduce/examplemodule/main.go
@@ -1,9 +1,8 @@
 package main

 import (
-	"fmt"
-
 	"examplemodule/foo"
+	"fmt"
 )

 func main() {
$ cd examplemodule/
$ golangci-lint run -c ../.golangci.yml
main.go:4:1: File is not properly formatted (gofumpt)
	"fmt"
^
1 issues:
* gofumpt: 1
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Hello,

This is the expected behavior.

- `golangci-lint run` can only be run inside a module.
- `golangci-lint fmt` is working per file, so don't care about modules.

This is basically the same difference as `go vet` and `go fmt`.

**Comment by nieomylnieja** [comment]
Gotcha, thank you for the swift answer :) 
Is this documented anywhere though, maybe its worthwhile mentioning this behavior? I haven't had any luck finding anything on that in the linter's docs.
